Add hover states for buttons

guergana opened the pull request:
- fixes #596 

Hover states done: 
- Upload your data sidebar :heavy_check_mark: 
- Upload your data empty screen :heavy_check_mark: 
- Create folder Dialog :heavy_check_mark: 
- Publish Dialog :heavy_check_mark: 
- File Upload Dialog - Add External Data Tab :heavy_check_mark:
- Launch Screen (aka Welcome Banner) :heavy_check_mark: 
- How the ODE saves your data :heavy_check_mark: 
- Publish button upper right bar (missing design) :heavy_check_mark: 
- Unsaved changes :heavy_check_mark: 
- Rename file/folder dialog :heavy_check_mark: 
- Delete dialog :heavy_check_mark:

roll commented:
Hi, looks good!

A few things I spotted:

The hover border doesn't seem to be from the palette:

![hover](https://github.com/user-attachments/assets/e05637a9-3e06-4c41-8afb-3e8e2d12ea34)

Publish button looks to be in the wrong place:

![publish](https://github.com/user-attachments/assets/92d899e6-1852-4aee-8a13-c8847d5eb44f)

Not related to this PR I think but the "Unsaved Changes" dialog is really unclear:

![changes](https://github.com/user-attachments/assets/a922e0a4-3114-42b0-b292-1f6f2b47e716)

Cancel the action or cancel the changes? Usually, in this context cancel means don't do anything and get me back (not discard)

guergana commented:
> Hi, looks good!
> 
> A few things I spotted:
> 
> The hover border doesn't seem to be from the palette:
> 
> ![hover](https://private-user-images.githubusercontent.com/557395/383469851-e05637a9-3e06-4c41-8afb-3e8e2d12ea34.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwMjQxMjEsIm5iZiI6MTczMTAyMzgyMSwicGF0aCI6Ii81NTczOTUvMzgzNDY5ODUxLWUwNTYzN2E5LTNlMDYtNGM0MS04YWZiLTNlOGUyZDEyZWEzNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwN1QyMzU3MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ODNjYzhiMGJjZGVkNmFhNmE0YWU5YWE1NjZhOTViOWY5MjcwNWM3MjgyZTYzMzU5MjM3YzlkZTU2NGFkYTc5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2SfQYbx7AfvXd4NKg27tdi3A5GF8IMM9A9DPSM4FHOk)
> 
> Publish button looks to be in the wrong place:
> 
> ![publish](https://private-user-images.githubusercontent.com/557395/383470003-92d899e6-1852-4aee-8a13-c8847d5eb44f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwMjQxMjEsIm5iZiI6MTczMTAyMzgyMSwicGF0aCI6Ii81NTczOTUvMzgzNDcwMDAzLTkyZDg5OWU2LTE4NTItNGFlZS04YTEzLWM4ODQ3ZDVlYjQ0Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwN1QyMzU3MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZjZkOWNkMjYxODBhYjczZGMwYjViOTRiNzJkMTIzOWE4YjlkM2I0MDNiMDY2ZDE1NTdlZGFiZDFlODgwMzM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5OLwPeNyE7JIG00RlAa9yOko4J6KDMIRArK4LvBbWJM)
> 
> Not related to this PR I think but the "Unsaved Changes" dialog is really unclear:
> 
> ![changes](https://private-user-images.githubusercontent.com/557395/383470506-a922e0a4-3114-42b0-b292-1f6f2b47e716.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwMjQxMjEsIm5iZiI6MTczMTAyMzgyMSwicGF0aCI6Ii81NTczOTUvMzgzNDcwNTA2LWE5MjJlMGE0LTMxMTQtNDJiMC1iMjkyLTFmNmYyYjQ3ZTcxNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwN1QyMzU3MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZjAyYjVlYjk2YTI3ZmQ5YzRiNWJkNGZhMjMzZjA0MDNhYjNhMDA3MmFiMjNhYzM4OWRhYjlhYzQ4MTlkOTFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.CW2cOM9QEO5cNEMXavEnddvyZ9uDZh2z3VZnnpS8KrM)
> 
> Cancel the action or cancel the changes? Usually, in this context cancel means don't do anything and get me back (not discard)

Hi @roll I have addressed your comments except the one from the cancel. What is your suggestion? To have the text say "discard" instead of cancel? We wanted to have a standard cancel action for all buttons, this is why we changed the discard text to have all say "cancel" but you are right, in this case is confusing. This should be discussed with @romicolman and @Faithkenny .

romicolman reviewed (approved):
Hi! I checked everything on Mac. All dialogs are now OK except from the Publish one. I tried to add an URL and the button never gets activated. I think this is a separate issue so if the button is aligned with the rest of the design, please, merge it and I'll create a separate ticket.

romicolman commented:
Hi! I have just checked all comments. Please, do not change the Cancel button for Discard. However, let's make the text clearer:

**Replace**

There are unsaved changes.

**For**

There are unsaved changes. Please, click save or cancel.

roll commented:
> Please, click save or cancel

Cancel action or changes?

romicolman commented:
Hi! Actions can be broad for users. Please @guergana let's procede with the text I suggested until we finish the user testing session. I'll inform you both if users mention find the change confusing.
